OrangeHRMTests/Tests/Tests.cs

```diff
@@ -0,0 +1,590 @@
+            const string UnvalidEmployeeName = "123456";
+
+            GenericPages.BaseTest.CreateEmployee();
+
+            GenericPages.BasePage.LeftMenuNavigationPanel.GoToLeftMenuItem(LeftMenuNavigationPanelButtonsNames.PIMButtonName);
+
+            GenericPages.PIMPage.EnterValueInInputTextField(InputFieldsNames.EmployeeNameInputFieldName, NewEmployeeName);
+            GenericPages.PIMPage.ClickDropdownListFirstPosition();
+            GenericPages.PIMPage.ClickSearchButton();
+
+            ClassicAssert.AreEqual(NewEmployeeName, Table.GetCellText("First (& Middle) Name"));
+
+            GenericPages.PIMPage.EnterValueInInputTextField(InputFieldsNames.EmployeeNameInputFieldName, Keys.Control + "a" + Keys.Delete);
+            GenericPages.PIMPage.EnterValueInInputTextField(InputFieldsNames.EmployeeNameInputFieldName, UnvalidEmployeeName);
+
+            ClassicAssert.AreEqual(PIMPageHeadersTextValues.EmployeeInformationHeaderTextValue, GenericPages.PIMPage.GetImployeeInformationText());
+
+            GenericPages.PIMPage.ClickSearchButton();
+
+            ClassicAssert.AreEqual(InfoMessageTextValues.NoRecordsFoundMessageText, GenericPages.PIMPage.GetPopUpMessageTextElement());
+
+            GenericPages.BaseTest.DeleteCreatedEmployee();
+        }
+
+        [Test]
+        public void N_EditEmployeeDetailsTest()
+        {
+            const string NewEmployeeName = "111 222";
+
+            GenericPages.BasePage.LeftMenuNavigationPanel.GoToLeftMenuItem(LeftMenuNavigationPanelButtonsNames.PIMButtonName);
+
+            GenericPages.BasePage.TopBarMenu.ClickTopbarMenuButtonByName(PIMPageTopBarMenuButtonsNames.EmployeeListTopBarMenuButtonName);
+            Table.ClickPencilEditButton();
+
+            ClassicAssert.AreEqual(PIMPageHeadersTextValues.PersonalDetailsHeaderTextValue, GenericPages.PIMPage.GetPersonalDetailsHeaderTextElement());
+
+            GenericPages.PIMPage.ClearFullUserName();
+            GenericPages.PIMPage.EnterFullUserName();
+            GenericPages.PIMPage.ClickSaveButton();
+            GenericPages.BasePage.TopBarMenu.ClickTopbarMenuButtonByName(PIMPageTopBarMenuButtonsNames.EmployeeListTopBarMenuButtonName);
+
+            ClassicAssert.AreEqual(PIMPageHeadersTextValues.EmployeeInformationHeaderTextValue, GenericPages.PIMPage.GetImployeeInformationText());
+
+            GenericPages.PIMPage.EnterValueInInputTextField(InputFieldsNames.EmployeeNameInputFieldName, NewEmployeeName);
+            GenericPages.PIMPage.ClickDropdownListFirstPosition();
+            GenericPages.PIMPage.ClickSearchButton();
+
+            ClassicAssert.AreEqual(NewEmployeeName, Table.GetCellText("First (& Middle) Name"));
+            ClassicAssert.AreEqual("333", Table.GetCellText(TableColumnsNames.LastNameTableColumnName));
+
+            GenericPages.BaseTest.DeleteCreatedEmployee();
+        }
+
+        [Test]
+        public void O_SearchAdminTest()
+        {
+            const string NewUserName = "111admin";
```

Comment:
the same value is used in the BaseTest, data is too hardcoded and not reliable. Because you create such a user by method from the BaseTest. If it won't be deleted in some other tests - your current test will fail what means test are not independent

I would better use data faker like Bogus for providing all this value. it's super easy to use, you may not to use hardcoded data at all

If don't agree or point why it's better is not clear for you - let's discuss

Reply:
Fixed